File: compDNA.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<string>
#include<format>
using namespace std;
string enterStrand();
string computeComp(string);
bool confirmStrand(string);
int main() {
   cout << "Compute Complementary - DNA\n";
   string DNA = enterStrand();
   string cDNA = computeComp(DNA);
   cout << "Here is your sequence: \n" << DNA << "\n";
}
bool confirmStrand(string DNA) {
   string dnaBases = "TACG";
   for(char i: DNA) {
      if (find(dnaBases.begin(),dnaBases.end(),i) == dnaBases.end())
      return false;
   };
   return true;
}
string enterStrand() {
   string DNA;
   while(true) {
      cout << "Enter DNA Strand\n";
      cin >> DNA;
      if (confirmStrand(DNA) == true) {
         return DNA;
      } else {
         cout << "Invalid Strand, Try Again\n";
      }
   }
}
string computeComp(string DNA) {
   return "Done";
}

File: compDNA.py
def enterStrand():
   while True:
      DNA = input("Enter DNA Strand\n").upper()
      if confirmStrand(DNA):
         return DNA
      else:
         print("Invalid Strand, Try Again")
         
def confirmStrand(DNA):
   for base in DNA:
      if base not in "ATGC":
         return False
   return True

def computeComp(DNA):
   cDNA = "3'- "
   for base in DNA:
      if base == "A":
         cDNA += "T"
      if base == "T":
         cDNA += "A"
      if base == "G":
         cDNA += "C"
      if base == "C":
         cDNA += "G"
   cDNA += " - 5'"
   return cDNA
         
def main():
   print("Compute Complementary - DNA")
   DNA = enterStrand()
   cDNA = computeComp(DNA)
   print(f"Here is your sequence: \n{cDNA}")
   